### PR TITLE
Bessiesteinberg/page branding

### PR DESCRIPTION
**Pull request by BessieSteinberg**
ENT-1933: Page Branding

- Add branding to Index and Program pages.
- Add panels in the wagtail CMS so that you can easily edit branding for those pages.
- Added ```/api/v1/pages/``` api so that you can query pages and their associated branding data. (shout out to @tuchfarber for is work on making that customized for use case)

### CODE REVIEW

File: designer/apps/pages/migrations/0006_auto_20190702_1505.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('pages', '0005_auto_20190701_1719'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='programpage',
+            name='uuid',
+            field=models.UUIDField(editable=False, unique=True),
```
Comment (tuchfarber) [inline designer/apps/pages/migrations/0006_auto_20190702_1505.py]: We're you able to save program pages with UUID being `editable=False`? I was going to remove the `editable=False` line in a followup PR.

File: designer/apps/pages/serializers.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+""" Serialzers for Page Models """
+from rest_framework import serializers
+from wagtail.wagtailcore.models import Page
+from designer.apps.pages.models import IndexPage, ProgramPage
+from designer.apps.branding.serializers import BrandingField
+
+
+class PageSerializer(serializers.ModelSerializer):
```
Comment (tuchfarber) [inline designer/apps/pages/serializers.py]: As per above, I think all the serializers should go in the `api` app.

File: designer/apps/api/v1/views.py [hunk 3]
```diff
@@ -5,29 +5,39 @@
...
+    "last_published_at",
+    "slug",
+    "title",
+    "type",
+    "uuid",
+)
+PAGE_API_FIELD_REPLACEMENT = {
+    "pk": "id"
+}
+# (INTERNAL_IDENTIFIER, EXTERNAL_IDENTIFIER) = ("pk", "id")
```
[the remark this answers is not in the document]
Comment (BessieSteinberg) [reply designer/apps/api/v1/views.py]: haha i'll get rid of it :D

File: designer/apps/branding/models.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+
+class Branding(models.Model):
+    """
+    Base Branding model
+    """
+    cover_image = models.ForeignKey(
+        'wagtailimages.Image',
+        null=True,
+        blank=False,
```
[the remark this answers is not in the document]
Comment (BessieSteinberg) [reply designer/apps/branding/models.py]: yes!

File: designer/apps/pages/models.py [hunk 5]
```diff
@@ -1,24 +1,51 @@
...
     body = RichTextField(blank=True)
     uuid = models.UUIDField(editable=False, unique=True)
 
     content_panels = Page.content_panels + [
         FieldPanel('body', classname="full"),
+        InlinePanel('branding', label="Program Page Branding", max_num=1),
     ]
+
+    def publish(self):
+        super(ProgramPageBranding, self).publish()
```
[the remark this answers is not in the document]
Comment (BessieSteinberg) [reply designer/apps/pages/models.py]: yeah i just deleted it, i had it in there for debugging purposes and it shouldn't be there.

File: designer/apps/api/v1/views.py [hunk 6]
```diff
@@ -42,11 +58,70 @@ def get_queryset(self):
...
+
+    def filter_by_hostname(self, queryset, hostname):
+        try:
+            site = Site.objects.get(hostname=hostname)
+        except Site.DoesNotExist:
+            # Return no pages since none match the passed in hostname
+            return queryset.none()
+        return queryset.descendant_of(site.root_page, inclusive=True)
+
+    def filter_unwanted_keys(self, pages):
```
[the remark this answers is not in the document]
Comment (BessieSteinberg) [reply designer/apps/api/v1/views.py]: yeah that makes sense - i'll make those changes

File: designer/apps/branding/serializers.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+""" Serializers for Branding Model """
+from rest_framework import serializers
+
+
+class BrandingField(serializers.PrimaryKeyRelatedField):
```
[the remark this answers is not in the document]
Comment (BessieSteinberg) [reply designer/apps/branding/serializers.py]: will do

File: designer/apps/pages/migrations/0006_auto_20190702_1505.py (hunk 1 above)
Comment (BessieSteinberg) [reply designer/apps/pages/migrations/0006_auto_20190702_1505.py]: So right now we can't edit 'uuid' and program pages can only be made via the management command. I am open to changing this, but if you already have a PR ready with that change we can wait for that. Which ever

File: designer/apps/pages/serializers.py (hunk 2 above)
Comment (BessieSteinberg) [reply designer/apps/pages/serializers.py]: will do

File: designer/apps/pages/migrations/0006_auto_20190702_1505.py (hunk 1 above)
Comment (tuchfarber) [reply designer/apps/pages/migrations/0006_auto_20190702_1505.py]: I'll follow up on it. No need to add more work here.

**Pull request merged**